### DISCUSSION TITLE
HTML Emails

### DISCUSSION THREAD

**Discussion by Kukks** [btcpayserver/btcpayserver#2229] [Ideas / Feature Requests]
Seems like our email designs have the quality of 💩 

Let's see what we can come up with. 

On the technical side of things, currently storing all the HTML and copy in C# vars  is not optimal. We should consider moving them to an html file which allows us to preview during dev easily. 

Additionally we can go slightly more hardcore and allow users to override the email template via UI. Maybe this can be a use-case for a plugin.

**Comment by dennisreimann** [comment]
A few notes for approaching this: Styling email content is a pain if you want to get fancy – I'd advise we do as much as we need to and not more than that.

Oftentimes email clients don't load external content and if we want to go with HTML emails we should

- inline css and images, so I think having a general email layout file the content gets wrapped with would be best
- keep it as streamlined as possible
- provide a plain text version too

**Reply by bolatovumar** [reply]
Yeah, this is true. Making a HTML email that works across many different email clients is very hard. If we end up shipping a HTML email template I would recommend using some existing template off the web.

**Comment by saliehendricks** [comment]
These are the components I outlined while doing a related PR. Something like: 

EmailLayout.html
```html
<html>
	<body style="all styles are inline">
		<table width="100%" cellspacing="0" cellpadding="0">
			<tr>
				<td>
					INLINE HEADER
				</td>
			</tr>
			<tr>
				<td>
					{MAIN_EMAIL_BODY_PLACEHOLDER}
				</td>
			</tr>
			<tr>
				<td>
					INLINE FOOTER
				</td>
			</tr>
		</table>	
	</body>
</html>
```

EmailCallToAction.html for basic button:
```html
<a style="" href="{action_link}">{action_name}</a>
```

Specific emails can then look like this:

PasswordReset.html
```html
Please reset you password <a href="">EmailHelper.CallToAction(name, link)</a>
```

SomeotherEmail.html ...

These can be later be editable via UI.

Note:
- This will read HTML files from disk multiple times unless some form of caching exists. May not be an issue, depends on load.
- Do we want to support dark mode emails?
- Magic strings are not great
